pkg/ruler/ruler.go

```diff
@@ -430,18 +449,147 @@ func (r *Ruler) newManager(ctx context.Context, userID string) (*promRules.Manag
 	}
 
 	// Wrap registerer with userID and cortex_ prefix
-	reg := prometheus.WrapRegistererWith(prometheus.Labels{"user": userID}, prometheus.DefaultRegisterer)
+	reg := prometheus.WrapRegistererWith(prometheus.Labels{"user": userID}, r.registry)
 	reg = prometheus.WrapRegistererWithPrefix("cortex_", reg)
-
+	logger := log.With(r.logger, "user", userID)
 	opts := &promRules.ManagerOptions{
 		Appendable:  tsdb,
 		TSDB:        tsdb,
 		QueryFunc:   promRules.EngineQueryFunc(r.engine, r.queryable),
 		Context:     user.InjectOrgID(ctx, userID),
 		ExternalURL: r.alertURL,
 		NotifyFunc:  sendAlerts(notifier, r.alertURL.String()),
-		Logger:      util.Logger,
+		Logger:      logger,
 		Registerer:  reg,
 	}
 	return promRules.NewManager(opts), nil
 }
+
+// GetRules retrieves the running rules from this ruler and all running rulers in the ring if
+// sharding is enabled
+func (r *Ruler) GetRules(ctx context.Context, userID string) ([]*rules.RuleGroupDesc, error) {
+	if r.cfg.EnableSharding {
+		return r.getShardedRules(ctx, userID)
+	}
+
+	return r.getLocalRules(userID)
+}
+
+func (r *Ruler) getLocalRules(userID string) ([]*rules.RuleGroupDesc, error) {
+	groupDescs := []*rules.RuleGroupDesc{}
+
+	var groups []*promRules.Group
+	r.userManagerMtx.Lock()
+	mngr, exists := r.userManagers[userID]
```

Comment:
We could compact it into the following, which also reduce the scope of variables:

```
if mngr, exists := r.userManagers[userID]; exists {
    groups = mngr.RuleGroups()
}
```

Reply:
👍